core/src/main/scala/tech/beshu/ror/accesscontrol/factory/JsonConfigStaticVariableResolver.scala

```diff
@@ -54,17 +54,31 @@ object JsonConfigStaticVariableResolver {
       .mapObject(_.mapValues(mapJson(_, errors)))
       .withString { str =>
         val resolved = NonEmptyString.unapply(str) match {
-          case Some(nes) => tryToResolveAllStaticSingleVars(nes, errors)
+          case Some(nes) =>
+            tryToResolveAllStaticSingleVars(nes, errors)
           case None => str
         }
-        resolvedStringToJson(resolved)
+        if (resolved =!= str)
+          resolvedStringToJson(resolved, json)
+        else json
       }
   }
 
-  private def resolvedStringToJson(resolvedStr: String) = {
+  private def preserveNumbersAsStrings(original: Json, newValue: Json) = {
+    {
+      for {
+        _ <- original.asString
+        newNum <- newValue.asNumber
+      } yield Encoder.encodeString(newNum.toString)
+    }.getOrElse(newValue)
```

Comment:
is it safe to "catch all exceptions" and return the `newValue` value? Maybe we should catch only a specific exception?

Reply:
Wait, asString and asNumber return Option, so either they "manage" to parse as the respective type, or they return None. What is there for us to  catch here? 🤔

Reply:
Ok, my bad